design/embedded-cluster-cli-driven-update.md

```diff
@@ -0,0 +1,278 @@
+
+#### Modified Update Command Behavior
+- **Conditional behavior**: Check `ENABLE_V3` environment variable to determine update mode
+- **V2 mode**: Current behavior (application airgap bundle updates)
+- **V3 mode**: Binary self-updating behavior
+- **Command signature (V3)**: `./binary update --license <license-file> [--version=<version>] [--force]`
+- **License requirement**: The `--license` flag is **required** when `ENABLE_V3=1` for authentication with replicated.app
+- **Validation**: Verify the binary can be updated (not installed via package manager)
+
+#### Command Options (V3 Mode)
+- `--license`: **Required** - License file for authentication with replicated.app
+- `--version`: Optional parameter to specify target version (defaults to latest)
+- `--force`: Skip confirmation prompts for automated scenarios
+
+#### Backward Compatibility
+- **Preserve V2 behavior**: When `ENABLE_V3` is not set or set to 0, maintain current airgap bundle update functionality
+
+### 2. Version Management Infrastructure
+
+**Current State:** Most version infrastructure already exists in the embedded cluster binary.
+
+**What's Already Available:**
+- Version information: `pkg/versions/versions.go` (EC version, k0s version)
+- Embedded release data: App slug, channel, version label via `pkg/release/release.go`
+- Version display: Existing `./binary version` command
+
+#### Architecture Approach
+
+**Dual Sequence Handling:**
+The embedded cluster must handle multiple versioning schemes simultaneously, each serving different purposes:
+
+- **Upstream cursors**: Channel sequence numbers from API (e.g., 1247, 1248, 1251, 1255)
+  - Used for API synchronization and incremental updates
+  - Can have gaps when releases are skipped or removed
+  - Channel-specific and may restart across different channels
+  - Essential for efficient API communication
+
+- **Version labels**: User-facing identifiers (e.g., v1.2.0, v1.2.1)
+  - What customers see and use in commands
+  - May follow semantic versioning or custom labeling schemes
+  - Used for `--version` flag and user communication
+  - Channel-agnostic when semver is enabled
+
+- **Embedded cluster versions**: Technical versions (e.g., 1.5.0+k8s-1.29)
+  - Indicates the embedded cluster infrastructure version
+  - Includes Kubernetes version and other technical details
+  - Used for compatibility checks and technical validation
+  - Independent of application versioning
+
+**Example Mapping:**
+```
+API Release          User Experience      Technical Details
+─────────────       ─────────────────     ─────────────────
+Cursor: 1247    →   Version: v1.2.0   →   EC: 1.5.0+k8s-1.29
+Cursor: 1248    →   Version: v1.2.1   →   EC: 1.5.0+k8s-1.29  
+Cursor: 1251    →   Version: v1.3.0   →   EC: 1.6.0+k8s-1.30
+```
+
+This multi-layered approach allows the binary to handle API efficiency, user experience, and technical compatibility as separate concerns.
+
+#### Version Comparison Logic
```

Comment:
i don't think this should be a thing in embedded cluster. all this logic should only live in replicated.app